### ISSUE TITLE
dCacheView authentication with macaroons

### ISSUE THREAD

**Issue by onnozweers** [dCache/dcache#5036]
Dear dCache devs,

How can I authenticate to dCacheView using a macaroon? I tried appending `?authz=<macaroon>` to the dCacheView URL, but that throws me into the username/password page. Trying with dCache 4.2.40.

I've found the "Add a shared file / Paste the macaroon here" dialogue, but I was wondering whether our users can allow someone without a dCache account to upload/download data with dCacheView.

Cheers,
Onno

**Comment by femiadeyemi** [comment]
Hi Onno,

First, you can generate macaroon in other to share files by using dCache-view. The advantage of this is that it gives you three different options:
1. shareable link: `<http|https>:<frontent-url>:<port number>/#!/shared-link?m=<macaroon>`
2. macaroon
3. qr code - generated based on the shareable link

Second, to view a shared files using dCache view is possible and yes, you don't need dCache account to view the shared files provided you have one of the 3 options above. Basically, to view a shared files, you can either use the shareable link or the macaroon directly. To use the macaroon, clicked on the orange tab on you left hand side, this is the shared-files section. At the bottom right corner you will see a circular float button with a plus icon, click on it. A dialog box will be shown to the user which the user can paste the macaroon inside. The alternative is to use the shareable link and it is the form above.

I hope this help.

Best regards,
Olufemi

**Comment by onnozweers** [comment]
Dear Olufemi,

Thanks for explaining the dCache View macaroon get URL format! I was able to create macaroons using the "Share file" dialogue. However, when I copy this link (with the `#!/shared-link?m=` in it) to a new private browser window, I am thrown into the username/password dialogue. I've tried in Firefox and Chrome on OSX, and Firefox on Centos7. I also tried the QR code on my iPhone, but it behaves the same: I get a username/password dialogue.

When I add macaroons on the "Shared files" page, they are added. But in case of a directory I can't view any objects in it (it says the directory is empty, while it contains two files). If it's a macaroon to share a file, I get a 403 error when I try to download it. I can download the file normally without macaroon.

I deserialized one of the macaroons, and I noticed that there were two identical `cid path:` statements. Is that expected? There is also a `cid home:` statement, which is new to me.

Kind regards,
Onno
